x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -532,6 +532,7 @@ private void followLeaderIndex(String autoFollowPattenName,
             request.setRemoteCluster(remoteCluster);
             request.setLeaderIndex(indexToFollow.getName());
             request.setFollowRequest(followRequest);
+            request.setWaitForCompletion(true);
```

Comment:
why set this to true?
This will mean that `recordLeaderIndexAsFollowFunction` will possibly be only called after a very long time, i.e., after the recovery from remote is completed /cc: @martijnvg